### ISSUE TITLE
Library loading for every known RIDs

### ISSUE THREAD

**Issue by Odonno** [Cysharp/csbindgen#80]
I have to admit that this is not up to you to propose this but it is really sad that the .NET SDK doesn't offer a built-in way to retrieve native libraries. Or even providing a proper example on how to load native libraries.

Anyway, there is already a great solution in the readme that can work with `win|osx|linux` OS and `x86|x64|arm64` process architectures. However, it does not cover the full list of RIDs, or at least the most commonly used you can find here: https://learn.microsoft.com/en-us/dotnet/core/rid-catalog#known-rids

The current implementation is sufficient to cover Windows and OSX platforms. However, I do not find a way to cover Android and iOS platforms. And same for deduping musl vs. non-musl.

Did you encounter this issue and have you found a solution?

**Comment by neuecc** [comment]
NativeLibrary.SetDllImportResolver dynamically determines the location of the library to import, so it is not bound by any specific RID and can be freely selected if the information is available in the application.

> On Linux, it may be the kernel version-to get the Linux distro name and version information, the recommended approach is to read the /etc/os -release file.

You may choose it yourself.

**Comment by Odonno** [comment]
I tried something by writing a console exe and defining a RID. I got really confused by the output.

But then:
```
The [.NET SDK flattens any directory structure](https://github.com/dotnet/sdk/issues/9643) under runtimes/{rid}/native/ when copying to the output directory.
```

This indeed flatten the runtimes directory so custom loading logic is not necessary. So, I assume it should work for every condition when you define a specific RID.

* For cloud native apps, it will use a fallback logic that goes `linux-musl-x64` -> `linux-x64` -> `linux` -> `unix` -> `any`
* For desktop apps, it will use a logic per the environment OS. For Windows, `win-x64` -> `win` -> `any`. This imply a bundle per OS/tuple OS-architecture
* For mobile apps, it will already have a predefined RID for iOS and Android. On Apple devices, I assume it is easily handled. Still unsure for the Android ecosystem.
* And it will work pretty much the same for other scenario I have not put interest in yet

I suppose I can keep this custom loading logic for cases where the developer does not define a RID. Not perfect but it should work.
